File: Data Structures & Algorithms/binary-tree-maximum-path-sum/submission-0.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func maxPathSum(root *TreeNode) int {
    globalMax := math.MinInt

    var dfs func(node *TreeNode) int
    dfs = func(node *TreeNode) int {
        if node == nil {
            return math.MinInt
        }

        left, right := max(0, dfs(node.Left)), max(0, dfs(node.Right))
        disjoinedPath := node.Val + left + right
        continuedPath := node.Val + max(left, right)

        globalMax = max(globalMax, max(disjoinedPath, continuedPath))

        return continuedPath
    }

    dfs(root)
    return globalMax
}

File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

from collections import deque

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if root is None:
            return 0

        queue = deque([root])
        depth = 0

        while len(queue) > 0:
            level_size = len(queue)
            for _ in range(level_size):
                top = queue.popleft()
                
                if top.left is not None:
                    queue.append(top.left)
                
                if top.right is not None:
                    queue.append(top.right)

            depth += 1

        return depth